tidy3d/components/base.py

```diff
@@ -727,12 +741,14 @@ def make_json_compatible(json_string: str) -> str:
             json_string = json_string.replace("Infinity", '"Infinity"')
             return json_string.replace(tmp_string, '"-Infinity"')
 
-        json_string = self.json(indent=INDENT, exclude_unset=False)
+        if "exclude_unset" not in kwargs:
```

Comment:
maybe in looking at this again, it could be better to just set these as kwargs in the function signature, like
```py
def _json_string(self) -> str:
    return self._json()

def _json(indent=INDENT, exclude_unset=False, **kwargs):
    ...
   
    json_string = self.json(indent=indent, exclude_unset=exclude_unset, **kwargs)
```
I think that will work and avoids unnecessary dict annoyances

Reply:
Didn't we discuss something like this once about some other method, and find that if e.g. `indent` is also in `kwargs`, it errors?

Reply:
Oh sorry this is different.

Reply:
yea there was something weird about this, but I think it works in this case. I tried it in a notebook. good to double check though.

Reply:
Updated as suggested.

Reply:
ok, you might need to pass indent and the other kwarg in the backend call to `_json()` too just FYI

Reply:
Well I'm fine with the defaults so no, right?

Reply:
oh right, it should be good yea